app/src/main/java/com/example/caregiver/ProfileInfo.java

```diff
@@ -0,0 +1,265 @@
+            public void onDataChange(DataSnapshot dataSnapshot) {
+                currentName = dataSnapshot.child("name").getValue().toString();
+                currentEmail = dataSnapshot.child("email").getValue().toString();
+                EditText nameField = (EditText) view.findViewById(R.id.profileName);
+                EditText emailField = (EditText) view.findViewById(R.id.profileEmail);
+                nameField.setHint(currentName);
+                emailField.setHint(currentEmail);
+
+            }
+            @Override
+            public void onCancelled(DatabaseError databaseError) {
+                Log.d("failure", "Unable to obtain user information");
+            }
+        });
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        // Get current userId
+        SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(getActivity());
+        String userId = preferences.getString("userId", "");
+
+        // Inflate the layout for this fragment
+        View view = inflater.inflate(R.layout.fragment_profile_info, container, false);
+        displayUserInfo(view, userId);
+
+        // Get button id, text fields id and set listeners
+        Button updateButton = (Button) view.findViewById(R.id.profileUpdateButton);
+        updateButton.setOnClickListener(updateUserInfoListener);
+        nameField = (EditText) view.findViewById(R.id.profileName);
+        emailField = (EditText) view.findViewById(R.id.profileEmail);
+        newPasswordField = (EditText) view.findViewById(R.id.profileNewPassword);
+        confirmPasswordField = (EditText) view.findViewById(R.id.profileNewPassword2);
+        errorMessage = (TextView) view.findViewById(R.id.profileInfoMessage);
+
+        // Set color
+        red = view.getResources().getColor(R.color.red);
+        green = view.getResources().getColor(R.color.green);
+
+        return view;
+    }
+
+    /**
+     * Render the error and success message field.
+     * @param sourceString The text message to be displayed.
+     * @param color The color for the text message (red for error, green for success).
+     */
+    public void displayMessage(String sourceString, int color) {
+        errorMessage.setText(Html.fromHtml(sourceString));
+        errorMessage.setVisibility(View.VISIBLE);
+        errorMessage.setTextColor(color);
+    }
+
+    /**
+     * Handle user password update.
+     * @param user The current user who is logged in the app
+     * @param newPassword User new password
+     * @param confirmPassword User new password (should be same as newPassword)
+     */
+    public void changePassword(
+            FirebaseUser user, @NonNull String newPassword, @NonNull String confirmPassword) {
+        if (!newPassword.equals(confirmPassword)) {
+            displayMessage("Your new and confirm password must be the same.", red);
```

Comment:
I think this error message is usually called "Passwords do not match."

Reply:
Fixed it in my latest PR.